catalog/cataloger_diff.go

```diff
@@ -194,56 +487,37 @@ func (c *cataloger) diffFromChild(ctx context.Context, tx db.Tx, childID, parent
 		PlaceholderFormat(sq.Dollar).
 		ToSql()
 	if err != nil {
-		return fmt.Errorf("effective commits sql: %w", err)
+		return nil, err
 	}
+	var effectiveCommits diffEffectiveCommits
 	err = tx.Get(&effectiveCommits, effectiveCommitsQuery, args...)
 	effectiveCommitsNotFound := errors.Is(err, db.ErrNotFound)
 	if err != nil && !effectiveCommitsNotFound {
-		return fmt.Errorf("select effective commit: %w", err)
+		return nil, err
 	}
 	if effectiveCommitsNotFound {
 		effectiveCommits.ChildEffectiveCommit = 1 // we need all commits from the child. so any small number will do
 		parentEffectiveQuery, args, err := psql.Select("commit_id as parent_effective_commit").
 			From("catalog_commits").
-			Where("branch_id = ? AND merge_source_branch = ?", childID, parentID).
+			Where("branch_id = ? AND merge_source_branch = ? AND merge_type = 'from_parent'", childID, parentID).
```

Comment:
Is this a bug-fix or a result of splitting a function?  If the former, can we add a test?

Reply:
@tzahij suggested being very explicit - there is no way you will match merge_source_branch with a different merge type between these branches.